cli/azd/internal/vsrpc/server_service.go

```diff
@@ -53,6 +55,14 @@ func (s *serverService) StopAsync(ctx context.Context) error {
 	// TODO(azure/azure-dev#3286): Need to think about how shutdown works. For now it is probably best to just have the
 	// client terminate `azd` once they know all outstanding RPCs have completed instead of trying to do a graceful
 	// shutdown on our end.
+
+	ts := telemetry.GetTelemetrySystem()
+	// Flush all in-memory telemetry data before stopping.
+	err := ts.Shutdown(ctx)
```

Comment:
Do we need to arrange for the gorouitine that is doing the upload to stop?

Reply:
Added support for cancellation.

In general, we don't lose much if we abruptly shut down the process. The telemetry items are largely persisted on disk, and only removed after successful transmission. The worst-case scenario is an item is reuploaded if the shutdown occurred after an item is transmitted but the item isn't removed.

I was slightly hesitant of increasing complexity. Shutdowns are tricky to be proven to be time-bounded, and usually bad shutdown behaviors lead to unpredictability to the consuming tool, eventually resulting in a degraded experience for the user. But given that @wbreza also voiced a similar suggestion, I added the few lines of code to support this.

I manually tested the shutdown scenario by simulating an infinite loop within the telemetry process, and observed proper cancellation.